src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,229 @@
+      return result;
+    }
+
+    int columnCountTarget = 0;
+    if (salesforceData.size() > 0) {
+      columnCountTarget = salesforceData.get(0).entrySet().size();
+    }
+    // Get the column count of the first JsonObject in bigQueryData
+    int jsonObjectIdx = 0;
+    int columnCountSource = 0;
+    if (bigQueryData.size() > 0) {
+      columnCountSource = bigQueryData.get(jsonObjectIdx).entrySet().size();
+    }
+    // Compare the number of columns in the source and target
+//    Assert.assertEquals("Number of columns in source and target are not equal",
+//                        columnCountSource, columnCountTarget);
+
+    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String tableName = "E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575";
+    // Build the table reference
+    TableId tableRef = TableId.of(projectId, dataset, tableName);
+
+    // Get the table schema
+    Schema schema = bigQuery.getTable(tableRef).getDefinition().getSchema();
+
+    // Iterate over the fields and print the column name and type
+    int currentColumnCount = 1;
+    while (currentColumnCount <= columnCountSource) {
+      for (Field field : schema.getFields()) {
+        String columnName = field.getName();
+        String columnType = field.getType().toString();
+        String columnTypeName = field.getType().getStandardType().name();
+        System.out.println("Column Name: " + columnName);
+        System.out.println("Column Type: " + columnType);
+        System.out.println("Column TypeName: " + columnTypeName);
+
+        switch (columnType) {
+
+          case "BOOLEAN":
+            boolean sfDateString = salesforceData.get(0).get(columnName).getAsBoolean();
+            boolean bqDateString = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+            Assert.assertEquals(sfDateString, bqDateString);
+            break;
+
+          case "FLOAT":
+            Double source = salesforceData.get(0).get(columnName).getAsDouble();
+            Double target = bigQueryData.get(jsonObjectIdx).get(columnName).getAsDouble();
+            Assert.assertTrue(String.format("Different values found for column: %s", columnName),
+                              source.compareTo(target) == 0);
+
+            break;
+
+          case "TIMESTAMP":
+            OffsetDateTime sourceDateTime = OffsetDateTime.parse(
+              salesforceData.get(0).get(columnName).getAsString(),
+              DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ"));
+            // Format the source timestamp in the desired format
+            String formattedSourceTimestamp = sourceDateTime.toString();
+            String formattedTargetTimestamp = bigQueryData.get(jsonObjectIdx).get(columnName).getAsString();
+            Assert.assertEquals(formattedTargetTimestamp, formattedSourceTimestamp);
```

Comment:
Please add error message in the Assertion.

Reply:
done